### ISSUE TITLE
minimum Instance requirement For ESGIFY model

### ISSUE THREAD

**Issue by Yeementh24** [sb-ai-lab/ESGify#1]
Hi, I am facing the problem with memory requirement for ESGIFY model to run.
Can u Guide me with the resource required for esgify model.

**Comment by E-Kovtun** [comment]
Hello, without Flair NER the model requires near 3 GB RAM. If you don't have GPU and use NER  it requires 4.5 GB RAM. If you use GPU and NER it requires 8 gb RAM and 2 GB RAM for NER. It also depends on the number of texts due to batch encoding. I can help you to solve the problem if you will send logs of error. We also have example in google colab. May be it can help you to  https://colab.research.google.com/drive/15YcTW9KPSWesZ6_L4BUayqW_omzars0l?usp=sharing

**Comment by Yeementh24** [comment]
Actually we are using amazon sagemaker notebook instance ml.g4dn.4xlarge we
have around 514 sentences when ever we are running through the model kernal
die even it happen same with google colab it run out of ram
Then we tried in batches like 100 sentences ,also we tried to decrease it
to 50 sentences but the kernal dies at every point

**Comment by kazakov15** [comment]
I recommend you add to you block of code that 2 strings:
model = model.eval()
with torch.no_grad():
&nbsp;&nbsp; to_model = tokenizer.batch_encode_plus(
&nbsp;&nbsp;                    texts,
                    &nbsp;&nbsp;add_special_tokens=True,
                    &nbsp;&nbsp;max_length=512,
                    &nbsp;&nbsp;return_token_type_ids=False,
                    &nbsp;&nbsp;padding="max_length",
                    &nbsp;&nbsp;truncation=True,
                    &nbsp;&nbsp;return_attention_mask=True,
                    &nbsp;&nbsp;return_tensors='pt',
                  &nbsp;&nbsp;)
&nbsp;&nbsp;results = model(**to_model)

**Comment by Yeementh24** [comment]
Hey Sb-Ai-Lab ,
I am writing to express my sincere gratitude for your amazing work on
creating the ML model that solved our working problem statement.
You have demonstrated exceptional skills and creativity in developing such
a powerful and elegant solution. I appreciate your dedication, passion, and
 innovation .
Thank you for sharing your brilliant work with us. I hope to learn from you
and collaborate with you in the future.
Sincerely,
Yeementh Virutkar
Machine Learning Engineer at Quantiphi

**Comment by Yeementh24** [comment]
Hi Sb-Ai-Lab
Hi we are facing problem in deployment of Esgify model on amazon sagemaker
can u provide the script for deploying ESGify model

**Comment by kazakov15** [comment]
@Yeementh24, please explain what do you mean under deploying script? Is file with requirements enough for you?  Or you want something else?

**Comment by Yeementh24** [comment]
Hi Sb-Ai-Lab/Esgify
I want to deploy Esgify model on amazon sagemaker environment as endpoint
such that i don’t need to load it on notebook as it acquire memory space on
notebook .
